robot/markers.py

```diff
@@ -46,6 +46,45 @@ def rot_y_degrees(self) -> Degrees:
         return Degrees(math.degrees(self.rot_y_radians))
 
 
+_Orientation = NamedTuple('Orientation', (
+    ('rot_x_radians', Radians),
+    ('rot_y_radians', Radians),
+    ('rot_z_radians', Radians),
+))
+
+
+class Orientation(_Orientation):
+    """
+    Represents the orientation in 3d space as rotations around x, y, and z axes.
+
+    Rotations around the different axes can be thought of as follows:
+      - X rotation represents pitch. Positive rotation
+      around the X axis represents a rotation with the nearest point going downwards,
+      and furthest going upwards. (i.e. a bowing motion is a positive x rotation)
+      - Y rotation represents yaw. Positive rotation around the Y axis represents
+      a rotation with the nearest point going to the left, and the furthest going to the
+      right.
+      - Z rotation represents roll. Positive rotation around the Z axis represents
+      a rotation with the highest point going to the right, and the bottom going to the
+      left.
```

Comment:
Z rotation is aligned with a clockwise rotation about the axis (when viewed from the positive end of the axis), yet the other rotations are counter-clockwise. Is the difference intentional?

I'm not sure whether this matters, just noting it as my assumption was that they'd be consistent.

Reply:
The Z axis actually comes towards you, so its still a counter-clockwise rotation around the axis.

Reply:
> The Z axis actually comes towards you,

You're right that I was wrong, though I'm not sure I agree with this statement -- positive Z is away from the camera.

> so its still a counter-clockwise rotation around the axis.

Indeed.

Reply:
Further to my being wrong here -- I think that it's `Y` which is actually clockwise; might this be a result of our y-coordinate being defined with a different positive direction to that of OpenCV?